ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoTransactionTreesSpec.scala

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.store.dao
+
+import com.digitalasset.daml.lf.transaction.Node.{NodeCreate, NodeExercises}
+import com.digitalasset.daml.lf.value.Value.AbsoluteContractId
+import com.digitalasset.ledger.EventId
+import com.digitalasset.platform.ApiOffset
+import org.scalatest._
+
+private[dao] trait JdbcLedgerDaoTransactionTreesSpec
+    extends OptionValues
+    with Inside
+    with LoneElement {
+  this: AsyncFlatSpec with Matchers with JdbcLedgerDaoSuite =>
+
+  behavior of "JdbcLedgerDao (lookupTransactionTreeById)"
+
+  it should "return nothing for a mismatching transaction id" in {
+    for {
+      (_, tx) <- storeCreateTransaction()
+      result <- ledgerDao.transactionsReader
+        .lookupTransactionTreeById(transactionId = "WRONG", Set(tx.submittingParty.get))
+    } yield {
+      result shouldBe None
+    }
+  }
+
+  it should "return nothing for a mismatching party" in {
+    for {
+      (_, tx) <- storeCreateTransaction()
+      result <- ledgerDao.transactionsReader
+        .lookupTransactionTreeById(tx.transactionId, Set("WRONG"))
+    } yield {
+      result shouldBe None
+    }
+  }
+
+  it should "return the expected flat transaction for a correct request (create)" in {
+    for {
+      (offset, tx) <- storeCreateTransaction()
+      result <- ledgerDao.transactionsReader
+        .lookupTransactionTreeById(tx.transactionId, Set(tx.submittingParty.get))
+    } yield {
+      inside(result.value.transaction) {
+        case Some(transaction) =>
+          val (eventId, createNode: NodeCreate.WithTxValue[AbsoluteContractId]) =
+            tx.transaction.nodes.head
+          transaction.commandId shouldBe tx.commandId.get
+          transaction.offset shouldBe ApiOffset.toApiString(offset)
+          transaction.transactionId shouldBe tx.transactionId
+          transaction.workflowId shouldBe tx.workflowId.getOrElse("")
+          val created = transaction.eventsById.values.loneElement.getCreated
+          transaction.rootEventIds.loneElement shouldEqual created.eventId
+          created.eventId shouldBe eventId
+          created.witnessParties should contain only tx.submittingParty.get
+          created.agreementText.getOrElse("") shouldBe createNode.coinst.agreementText
+          created.contractKey shouldBe None
+          created.createArguments shouldNot be(None)
+          created.signatories should contain theSameElementsAs createNode.signatories
+          created.observers should contain theSameElementsAs createNode.stakeholders.diff(
+            createNode.signatories)
+          created.templateId shouldNot be(None)
+      }
+    }
+  }
+
+  it should "return the expected flat transaction for a correct request (exercise)" in {
```

Comment:
This test reads the transaction tree, not the flat transaction.

Reply:
Yes, thanks, copy-pasta.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5089/commits/eba57bf61d9a7587aa53a066fc83b479c6980983